irc/client.go

```diff
@@ -627,20 +620,28 @@ func (client *Client) HasUsername() bool {
 }
 
 // SetNames sets the client's ident and realname.
-func (client *Client) SetNames(username, realname string) error {
-	// do this before casefolding to ensure these are actually ascii
+func (client *Client) SetNames(username, realname string, fromIdent bool) error {
+	limit := client.server.Config().Limits.IdentLen
+	if !fromIdent {
+		limit -= 1 // leave room for the prepended ~
+	}
+	if len(username) > limit {
```

Comment:
could we maybe switch the argument order here? personal preference, I like having the larger argument on the right always (keeps smaller on left and larger on right in all these sorta comparisons). not for here at all but might do a pass over the project generally sometime and make that the default as well

Reply:
Just checking: `limit < len(username)`?

Reply:
yepyep, that'd be perfect thanks